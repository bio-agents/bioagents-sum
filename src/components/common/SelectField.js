import React from 'react'
import { Col, ControlLabel, FormControl, FormGroup } from 'react-bootstrap'

const SelectField = ({ label, children, input }) => (
  <FormGroup controlId={input.name}>
    <Col componentClass={ControlLabel} sm={2}>
      {label}
    </Col>
    <Col sm={10}>
      <FormControl componentClass='select' {...input}>
        {children}
      </FormControl>
    </Col>
  </FormGroup>
)

export default SelectField
