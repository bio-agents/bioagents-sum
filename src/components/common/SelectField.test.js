import React from 'react'
import renderer from 'react-test-renderer'
import SelectField from './SelectField'

it('Renders SelectField correctly', () => {
  const component = renderer.create(
    <SelectField label='Please, select something' input={{ name: 'snapshot-select' }}>
      <option value='option1'>{'Option 1'}</option>
      <option value='option2'>{'Option 2'}</option>
      <option value='option3'>{'Option 3'}</option>
    </SelectField>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
