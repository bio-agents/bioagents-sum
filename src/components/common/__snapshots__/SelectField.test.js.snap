exports[`test Renders SelectField correctly 1`] = `
<div
  className="form-group">
  <label
    className="col-sm-2 control-label"
    htmlFor="snapshot-select">
    Please, select something
  </label>
  <div
    className="col-sm-10">
    <select
      className="form-control"
      id="snapshot-select"
      name="snapshot-select"
      type={undefined}>
      <option
        value="option1">
        Option 1
      </option>
      <option
        value="option2">
        Option 2
      </option>
      <option
        value="option3">
        Option 3
      </option>
    </select>
  </div>
</div>
`;
